controllers/component_dependency_update_controller.go

```diff
@@ -351,17 +348,35 @@ func (r *ComponentDependencyUpdateReconciler) handleCompletedBuild(ctx context.C
 	for i := range components.Items {
 		comp := components.Items[i]
 		if slices.Contains(updatedComponent.Spec.BuildNudgesRef, comp.Name) {
-			toUpdate = append(toUpdate, comp)
+			componentsToUpdate = append(componentsToUpdate, comp)
 		}
 	}
 	var nudgeErr error
-	immediateRetry, nudgeErr = r.UpdateFunction(ctx, r.Client, r.Scheme, r.EventRecorder, toUpdate, &BuildResult{BuiltImageRepository: repo, BuiltImageTag: tag, Digest: digest, Component: updatedComponent, DistributionRepositories: distibutionRepositories, FileMatches: nudgeFiles})
 
-	if nudgeErr != nil {
+	var targets []updateTarget
+	newTargets := r.ComponentDependenciesUpdater.GetUpdateTargetsGithubApp(ctx, componentsToUpdate)
+	log.Info("found new targets for git app", "targets", len(newTargets))
+	if len(newTargets) > 0 {
+		targets = append(targets, newTargets...)
+	}
+
+	newTargets = r.ComponentDependenciesUpdater.GetUpdateTargetsBasicAuth(ctx, componentsToUpdate)
+	log.Info("found new targets for basic auth", "targets", len(newTargets))
+	if len(newTargets) > 0 {
+		targets = append(targets, newTargets...)
+	}
+
+	if len(targets) > 0 {
+		log.Info("will create renovate job")
+		nudgeErr = r.ComponentDependenciesUpdater.CreateRenovaterPipeline(ctx, pipelineRun.Namespace, targets, true, &BuildResult{BuiltImageRepository: repo, BuiltImageTag: tag, Digest: digest, Component: updatedComponent, DistributionRepositories: distibutionRepositories, FileMatches: nudgeFiles})
```

Comment:
Maybe we can define `BuildResult` in a separate line? It would be easier to read